### PR TITLE
Angular

### PR DESCRIPTION
**Pull request by Bielbiel**

### CODE REVIEW

File: app/controllers/ctrlValidate.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+(function (angular)
+{
+    'use strict';
+    var app = angular.module('form-example1', []);
+
+    var INTEGER_REGEXP = /^-?\d+$/;
+    app.directive('integer', function ()
+    {
+        return {
+            require: 'ngModel-options', link: function (scope, elm, attrs, ctrl)
```
Comment (aniaw) [inline app/controllers/ctrlValidate.js]: I didn't see this before. Where did you find this?  `require: 'ngModel-options'` ?

File: app/controllers/ctrlValidate.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            {
+                ctrl.$validators.integer = function (modelValue, viewValue)
+                {
+                    if (ctrl.$isEmpty(modelValue)) {
```
Comment (aniaw) [inline app/controllers/ctrlValidate.js:14]: You can simplify this `if-s` and create only one

File: app/controllers/mainCtrl.js [hunk 3]
```diff
@@ -9,54 +9,74 @@
         ctrl.message = 'Trener Cinkciarza';
 
         ctrl.data = {
-            model: null, availableOptions: [{name: 'USD'}, {name: 'EUR'}, {name: 'CHF'}, {name: 'GBP'}, {name: 'CAD'}]
```
Comment (aniaw) [inline app/controllers/mainCtrl.js:12]: what means `data` - then name of this variable is unintuitive

File: app/controllers/mainCtrl.js [hunk 4]
```diff
@@ -9,54 +9,74 @@
...
-                        ctrl.moneyInSell = result.rates[0].bid;
-
-                        ctrl.buy = parseFloat(ctrl.amountForBay * ctrl.money).toFixed(2);
-                        ctrl.sell = parseFloat(ctrl.amountForSell * ctrl.moneyInSell).toFixed(2);
-                    });
+            if (ctrl.amount === undefined) {
+                window.alert('Zbyt duza lub zbyt mala kwota startowa');
+            } else {
+                ctrl.wallet.pln = ctrl.amount;
+                ctrl.wallet.EUR = 0;
```
Comment (aniaw) [inline app/controllers/mainCtrl.js:21]: Can you use here object form?
```
ctrl.wallet = {
EUR: 0,
USD:0,
...
```

File: app/controllers/mainCtrl.js [hunk 5]
```diff
@@ -9,54 +9,74 @@
...
+                                window.alert('Zbyt duza lub zbyt mala kwota startowa');
+                            } else if (Math.round(ctrl.wallet.pln * 100) < Math.round(ctrl.buyInWindow * 100)) {
+                                window.alert('Za malo pieniedzy :(');
+                            } else if (Math.round(ctrl.wallet.pln * 100) >= Math.round(ctrl.buyInWindow * 100)) {
+                                ctrl.wallet[ctrl.data.model] += ctrl.amountForBay;
+                                ctrl.wallet.pln -= ctrl.money * ctrl.amountForBay;
+                            }
+                        };
+
+                        ctrl.seller = function ()
```
Comment (aniaw) [inline app/controllers/mainCtrl.js]: You have to be consistent in naming. If you create `buy`function you should create `sell` function

File: app/views/main.html [hunk 6]
```diff
@@ -4,45 +4,96 @@
...
                 </table>
+            </div>
+        </div>
+    </div>
 
+    <div class="col-md-5">
+        <div class="panel panel-default">
+            <div class="panel-heading"><h2>Stan srodkow</h2></div>
+            <div class="panel-body">
+                <table>
```
Comment (aniaw) [inline app/views/main.html:58]: This table look werid. Try to create table in this way:

```
<table border="1" width="100%">
        <thead>
                <tr>
                        <th>Row 1, cell 1</th>
                        <th>Row 1, cell 2</th>
                        <th>Row 2, cell 3</th>
                </tr>
        </thead>
        <tbody>
                <tr>
                        <td>Row 2, cell 1</td>
                        <td>Row 2, cell 2</td>
                        <td>Row 2, cell 3</td>
                </tr>
        </tbody>
</table>
```
with `th` in `thead` and `tr` in `tbody`.

I think you should try to use here ng-repeat.

File: app/views/main.html [hunk 7]
```diff
@@ -63,43 +114,51 @@ <h2 class="nav-start">{{mainCtrl.message}}</h2>
...
-                                <p>Za <span id="plnViewInSell">{{mainCtrl.amountForSell}}</span> Otrzymasz <span id="oldWaluteForBay">{{mainCtrl.sell}}</span> PLN</p>
-                                <button class="btn btn-default" type="button" id="saveInSell">Zatwierdz</button>
+                            <div class="lastMargin">
+                                <p>Za <span id="plnViewInSell">{{mainCtrl.amountForSell}}</span> Otrzymasz <span id="oldWaluteForBay">{{mainCtrl.sell}}</span>
+                                    PLN</p>
                             </div>
                         </div>
                     </div>
-                </div>
+                </div>git
```
Comment (aniaw) [inline app/views/main.html]: `git`? :P

**Review by aniaw — CHANGES_REQUESTED** [review]

File: app/controllers/mainCtrl.js (hunk 4 above)
Comment (Bielbiel) [reply app/controllers/mainCtrl.js:21]: I cant make this becouse function not found

File: app/controllers/ctrlValidate.js (hunk 1 above)
Comment (Bielbiel) [reply app/controllers/ctrlValidate.js]: Me too but this work :)

**Pull request merged**